src/dmd/mtype.d

```diff
@@ -4743,14 +4743,40 @@ extern (C++) final class TypeFunction : TypeNext
                                 m = MATCH.exact;
                             else
                             {
-                                m = MATCH.nomatch;
                                 if (pMessage)
                                 {
+                                    /* https://issues.dlang.org/show_bug.cgi?id=22202
+                                     *
+                                     * If a function was deduced by semantic on the CallExp,
+                                     * it means that resolveFuncCall completed succesfully.
+                                     * Therefore, there exists a callable copy constructor,
+                                     * however, it cannot be called because scope constraints
+                                     * such as purity, safety or nogc.
```

Comment:
I would mention that `nothrow` is absent because copy constructors can't throw

Reply:
Copy constructors can throw, but nothrow is treated differently and does not suffer from this bug.

Reply:
Oh, then I misremembered